In JUnit, assertThrows is used to test if a specific exception is thrown by a piece of code. This is useful when you want to verify that certain actions result in expected exceptions, helping ensure your program handles errors as intended.

So it checks whether the method does return the given exception or not. If yes, the test case becomes true, otherwise false.

Imagine we have "Example" class where there is a method which returns RuntimeException and in testing method we will check it like that:

class ExampleTestThrowing {
    @Test
    void testThrowing(){
        Example e = new Example();
        assertThrows(RuntimeException.class, ()->e.sendThrowMessage());
    }
}

We need to use lambda expression here.

